File: 백준/Gold/11729. 하노이 탑 이동 순서/하노이 탑 이동 순서.cc
#include <iostream>
#include <cmath>
using namespace std;

void hanoi(int n, int from, int to) {
    int res;
    
    if(n == 1) cout << from << ' ' << to << '\n';
    else {
        res = 6 - from - to;
        hanoi(n-1, from, res);
        cout << from << ' ' << to << '\n';
        hanoi(n-1, res, to);
    }
}

int main() {
    int n, k;
    cin >> n;
    
    k =  pow(2,n) - 1;
    
    cout << k << '\n';
    hanoi(n, 1, 3);
}

File: 백준/Gold/11729. 하노이 탑 이동 순서/하노이 탑 이동 순서.swift
import Foundation

let n = Int(readLine()!)!

func hanoi(n: Int, start: String, to: String, via: String) {
    if n == 1 {
        print(start, to)
        return
    }
    
    hanoi(n: n - 1, start: start, to: via, via: to)
    hanoi(n: 1, start: start, to: to, via: via)
    hanoi(n: n - 1, start: via, to: to, via: start)
}

print(pow(2, n) - 1)
hanoi(n: n, start: "1", to: "3", via: "2")